hudi-client/hudi-client-common/pom.xml

```diff
@@ -193,6 +193,12 @@
     </dependency>
 
     <!-- Test -->
+    <dependency>
+      <groupId>org.apache.hudi</groupId>
+      <artifactId>hudi-tests-common</artifactId>
+      <version>${project.version}</version>
+      <scope>test</scope>
+    </dependency>
     <dependency>
       <groupId>org.junit.jupiter</groupId>
```

Comment:
We should extract these deps as well to "hudi-tests-common". while we're at it

Reply:
Do you know why junit-vintage-engine is being used by any chance? Do we still need to support running junit4 tests in some modules?

I can add in the junit-jupiter-api, junit-jupiter-engine to the common test module

Reply:
Good question. We can try to remove it and see what happens.

Reply:
still an exception in `TestFSUtils` where import org.junit.Rule; is used to inject env var, where it's not doable in junit 5, last time i checked.